### PR TITLE
Durable patterns migrated to isolated-process

### PR DESCRIPTION
**Pull request by piotrek-appstream**
Todo:
- [x] unit tests

### CODE REVIEW

File: src/AppStream.DurablePatterns/Steps/ConfigurationValidator/StepValidator.cs [hunk 1]
```diff
@@ -14,15 +14,18 @@ public void Validate(Step stepConfiguration, Step? previousStepConfiguration)
                 return;
             }
 
-            var previousStepResultType = previousStepConfiguration.PatternActivityResultType;
-            var currentStepInputType = stepConfiguration.PatternActivityInputType;
+            var previousStepResultType = Type.GetType(previousStepConfiguration.PatternActivityResultTypeAssemblyQualifiedName)!;
```
Comment (lukasz-appstream) [inline src/AppStream.DurablePatterns/Steps/ConfigurationValidator/StepValidator.cs:17]: Why is it better than providing type itself? The serialization is the problem? Size of it?

File: samples/AppStream.DurablePatterns.Samples.CombinedOrchestrator/Program.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using AppStream.DurablePatterns.Samples.CombinedOrchestrator.Activities;
+using AppStream.DurablePatterns.Samples.CombinedOrchestrator.Repository;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System.Threading.Tasks;
+
+namespace AppStream.DurablePatterns.Samples.CombinedOrchestrator
+{
+    public class Program
```
[the remark this answers is not in the document]
Comment (piotrek-appstream) [reply samples/AppStream.DurablePatterns.Samples.CombinedOrchestrator/Program.cs:9]: That's the way for isolated-process: https://nicksnettravels.builttoroam.com/migrate-azure-function/ (ctrl+f "program.cs")

File: src/AppStream.DurablePatterns/DurablePatternsServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -23,15 +23,15 @@ public static class DurablePatternsServiceCollectionExtensions
...
-                .AddSingleton<IDurablePatterns, Builder.DurablePatterns>()
-                .AddSingleton<IPatternActivityContractResolver, PatternActivityContractResolver>()
-                .AddSingleton<IStepValidator, StepValidator>()
-                .AddSingleton<IDurablePatternsExecutor, DurablePatternsExecutor>()
-                .AddSingleton<IStepExecutorFactory, StepExecutorFactory>()
-                .AddSingleton<IPatternActivityFactory, PatternActivityFactory>()
-                .AddSingleton<ActivityFunctionStepExecutor>()
-                .AddSingleton<FanOutFanInStepExecutor>()
-                .AddSingleton<IFanOutFanInOptionsValidator, FanOutFanInOptionsValidator>();
+                .AddTransient<IDurablePatterns, Builder.DurablePatterns>()
```
[the remark this answers is not in the document]
Comment (piotrek-appstream) [reply src/AppStream.DurablePatterns/DurablePatternsServiceCollectionExtensions.cs]: Idk I guess both should be fine, changed to scoped

File: samples/AppStream.DurablePatterns.Samples.CombinedOrchestrator/Program.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukasz-appstream) [reply samples/AppStream.DurablePatterns.Samples.CombinedOrchestrator/Program.cs:9]: yep, saw it later

File: src/AppStream.DurablePatterns/DurablePatternsServiceCollectionExtensions.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (lukasz-appstream) [reply src/AppStream.DurablePatterns/DurablePatternsServiceCollectionExtensions.cs]: Like scope seems like best memory/cpu wise as they don't require GC to work that much...? So if they work seems like a default, right?

File: src/AppStream.DurablePatterns/Steps/ConfigurationValidator/StepValidator.cs (hunk 1 above)
Comment (lukasz-appstream) [reply src/AppStream.DurablePatterns/Steps/ConfigurationValidator/StepValidator.cs:17]: That's stupid, but well, what can you do

**Review by lukasz-appstream — APPROVED** [review]

**Pull request merged**